ConstantReminders.Api.Tests/ConstantReminders/Services/UserServiceTests.cs

```diff
@@ -0,0 +1,223 @@
+                    Email = "john@test.com", AuthOId = "1", CreatedBy = "user", UpdatedBy = "user" },
+                new User { Id = Guid.NewGuid(), FirstName = "Jane", LastName = "Smith", PhoneNumber = "444-444-4444",
+                    Email ="jane@test.com", AuthOId ="2", CreatedBy = "user", UpdatedBy = "user" }
+
+            };
+
+            mockRepo.List().Returns(users);
+
+            var response = await service.GetUsers();
+
+            await mockRepo.Received(1).List();
+
+            Assert.NotNull(response);
+            Assert.Equal(2, response.Count);
+            Assert.Equal("John", response[0].FirstName);
+            Assert.Equal("Jane", response[1].FirstName);
+
+        }
+
+        [Fact]
+        public async Task GetUserById_ShouldReturnUser()
+        {
+     
+            var mockRepo = Substitute.For<IBaseRepository<User>>();
+            var service = new UserService(mockRepo);
+
+            var userId = Guid.NewGuid();
+
+            var user = new User 
+            {
+                Id = userId,
+                FirstName = "John",
+                LastName = "Doe",
+                PhoneNumber = "555-555-5555",
+                Email = "john@test.com",
+                AuthOId = "1",
+                CreatedBy = "user",
+                UpdatedBy = "user"
+            };
+
+       
+            mockRepo.GetByIdAsync(userId).Returns(user);
+
+            var response = await service.GetUserById(userId);
+
+            await mockRepo.Received(1).GetByIdAsync(userId);
+
+            Assert.NotNull(response);
+            Assert.Equal(userId, response.Id);
+            Assert.Equal("John", response.FirstName);
+        }
+
+        [Fact]
+        public async Task UpdateUser_ShouldUpdateAndReturnUser()
+        {
+    
+            var mockRepo = Substitute.For<IBaseRepository<User>>();
+            var service = new UserService(mockRepo);
+
+   
+            var userId = Guid.NewGuid();
+            var existingUser = new User { 
+                Id = userId, 
+                FirstName = "John",
+                LastName = "Doe",
+                PhoneNumber = "555-555-5555",
+                Email = "john@test.com",
+                AuthOId = "1",
+                CreatedBy = "user",
+                UpdatedBy = "user"
+            };
+
+           
+            var updatedUser = new User
+            {
+                Id = userId,
+                FirstName = "John",
+                LastName = "Updated",
+                PhoneNumber = "123456789",
+                Email = "john.updated@email.com",
+                AuthOId = "auth0-id",
+                UpdatedBy = "User",
+                CreatedBy = "User",
+                UpdatedDateTime = DateTime.Now
+            };
+
+       
+            mockRepo.GetByIdAsync(userId).Returns(existingUser);
+
+          
+            mockRepo.UpdateAsync(Arg.Any<User>()).Returns(Task.CompletedTask);  
+
```

Comment:
This test is failing. You can't get your code in if one of your tests is failing. UpdateAsync doesn't return a Completed Task, it returns a user, so you need to change what this mocked function returns and keep at it until the test passes.